### PR TITLE
BDTT-30

### PR DESCRIPTION
**Pull request by prussianman**
# Description

Fixes [BDTT-30](https://bottlenose.atlassian.net/browse/BDTT-30)
- [x] New feature (non-breaking change which adds functionality)

List all acceptance criterias of user story and indicate which has been solved previously and which is solved currently

 - [x] Retrieve shipping info via BL number
 - [x] Retrieve shipping info from via Container number
 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature work
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: webscrapers/creds/goodrich.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+username = ""
```
Comment (pandaponn) [inline webscrapers/creds/goodrich.py:1]: Is it necessary to merge this file into main?

**Review by pandaponn — CHANGES_REQUESTED** [review]
Return JSON responses instead of printing to console and find elements by ID instead of XPATH.

File: webscrapers/goodrich.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    # for executing javascript
+    action = ActionChains(driver)
+    
+    # landing page
+    driver.get("https://goodrich.co/track.php#new_tab")
+    # print("reachedLandingSite")
+
+    # iframe switch
+    try:
+        driver.switch_to.frame(driver.find_element(by=By.XPATH, value="/html/body/div[1]/iframe"))
```
[the remark this answers is not in the document]
Comment (prussianman) [reply webscrapers/goodrich.py:69]: When selecting by XPATH,

if there is an id, it seems like the XPATH generated will reference the element via id
```
            driver.find_element(by=By.XPATH, value="//*[@id=\"BlNo\"]").send_keys(tracking_identifier)
```

In cases where there isn't an id, either mix references will be used (id and then tags)
```
            driver.find_element(by=By.XPATH, value=" //*[@id=\"TAB1\"]/div[1]/div/div[3]/input").click()
 ```   
 or (just via tags :( )
``` 
           driver.switch_to.frame(driver.find_element(by=By.XPATH, value="/html/body/div[1]/iframe"))
```  
So I think it is fine to use XPATH? what do you think? For this particular iframe, there isn't an id

File: webscrapers/goodrich.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+from selenium.webdriver.common.by import By
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver import ActionChains
+
+#exceptions from selenium 
+from selenium.common.exceptions import NoSuchElementException
+
+# driver management (either first 2 or third)
+from selenium.webdriver.chrome.service import Service
+from webdriver_manager.chrome import ChromeDriverManager
```
[the remark this answers is not in the document]
Comment (prussianman) [reply webscrapers/goodrich.py:13]: I think there isn't any difference, apart from not needing to specify chromedriver path in the code

File: webscrapers/creds/goodrich.py (hunk 1 above)
Comment (prussianman) [reply webscrapers/creds/goodrich.py:1]: nope, there isnt

File: webscrapers/goodrich.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (prussianman) [reply webscrapers/goodrich.py:69]: Oh, I just searched, By.ID is faster, I'll make these changes

**Review by pandaponn — APPROVED** [review]
Great work here @prussianman, thanks for making the changes so quick.

**Pull request merged**